nixos/modules/i18n/input-method/fcitx5.nix

```diff
@@ -0,0 +1,33 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+
+let
+  im = config.i18n.inputMethod;
+  cfg = im.fcitx5;
+  fcitx5Package = pkgs.fcitx5.override { inherit (cfg) addons; };
+in
+  {
+    options = {
+      i18n.inputMethod.fcitx5 = {
+        addons = mkOption {
+          type = with types; listOf package;
+          default = [];
+          description = ''
+            Enabled Fcitx5 addons.
+          '';
+        };
+      };
+    };
+
+    config = mkIf (im.enabled == "fcitx5") {
+      i18n.inputMethod.package = fcitx5Package;
+
```

Comment:
Maybe need a `environment.profileRelativeSessionVariables` entry for env var `FCITX_ADDON_DIRS` to load any fcitx5 plugin.

```
environment.profileRelativeSessionVariables = {
  FCITX_ADDON_DIRS = [ "/lib/fcitx5" ];
};
```

Reply:
I've added the combined `/usr/share` directory to `XDG_DATA_DIRS` in the [wrapper](https://github.com/poscat0x04/nixpkgs/blob/9647e8ccc4a332a54541a9200f2cbf00ea021a16/pkgs/tools/inputmethods/fcitx5/wrapper.nix#L10-L13). It seems to be working with `fcitx5-rime`. Not sure if `FCITX_ADDON_DIRS` is also needed.

Reply:
I think that is indeed [enough](https://github.com/fcitx/fcitx5/blob/5db1e2fb749e8f9123c6d5ec1fa3537ec9f1e7a5/src/lib/fcitx-utils/standardpath.cpp#L83-L95)